contributor-guide/modules/ROOT/pages/testing/intro.adoc

```diff
@@ -1,212 +1,48 @@
-= Testing
-:idprefix:
-:idseparator: -
+= Introduction to Testing
+:navtitle: Introduction
 
-*Legacy content, currently being updated.*
+The Boost libraries are intended to be both reliable and portable. Every experienced programmer knows that means each library must be tested against a suitable number of test cases, on a range of platforms and compilers, and then tested again (_regression tested_) every time a change is made and also before every public release.
 
-The Boost libraries are intended to be both reliable and
- portable. Every experienced programmer knows that means each
- library must be tested against a suitable number of test cases,
- on a wide range of platforms, and then tested again (regression
- tested) every time a change is made and before every
- release.
+There are three broad dimensions to Boost library testing:
 
+. The actual test codes - unit tests, regression tests, new feature and bug tests. Refer to xref:testing/writing-tests.adoc[].
+. Tests that are integrated with the primary build systems, xref:build-systems/b2.adoc[] and xref:build-systems/cmake.adoc[].
+. xref:testing/continuous-integration.adoc[] tests.
 
-"Quality assurance based on a wide range of targeted tests"
- as one of the key answers to C.A.R Hoare's question "How did
- software get so reliable without proof."
+The new library developer needs to consider all three, however the initial focus on a new library will be the first in this list. Or, to put it another way, "Quality assurance based on a wide range of targeted tests" as one of the key answers to Professor Hoare's question:
 
+"_How did software get so reliable without proof._"
 
-== Regression test
+== Compiler Considerations
 
+Boost libraries rely on modern pass:[C++] features such as templates and the pass:[C++] Standard Library. Most modern compilers support those major features fairly well. But even today, years after the adoption of the pass:[C++] Standard, some compilers still don't support important minor features like partial template specialization.
 
-Boost uses an automatic [regression
- test suite](/doc/tools/regression/index.html) which generates HTML [compiler status
- tables](/development/testing.html#RegressionTesting).
+Boost library authors often expend a great deal of effort trying to work around compiler deficiencies. Nevertheless, some libraries will not compile at all with certain compilers, or may have reduced or compromised functionality. Even if the current release of a compiler supports a Boost library, older versions of the compiler may produce code that does not work properly.
 
+Boost releases are run through regression tests which automatically generates compiler status HTML tables for various platforms. Unless otherwise indicated, the pass:[C++] Standard Library implementation is the one shipped with the compiler. Refer to https://www.boost.org/doc/regression/library_status/doc/library_status.html[Generating Library Status Tables].
```

Comment:
> shipped with the compiler

This might be OK. But I think it would be technically the implementation available in the environment, because clang uses the GCC standard library on linux and we tend to use that in tests. That's actually kind of problem we have because we can use each clang version with a number of standard library implementations, which creates an exponential number of possibilities are can't really test.